Source/SteveSloth/SlingshotAmmoBaseClass.h

```diff
@@ -0,0 +1,68 @@
+/****************************************************************************************
+* Copyright: SteveSloth
+ * Name: Brandon Thomas
+ * Script: SlingshotAmmoBaseClass.h
+ * Date: April 23. 2024
+ * Description: This is the base header class for all ammo types
+ * TODO:
+ * Known Bugs:
+ ****************************************************************************************/
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Components/ActorComponent.h"
+#include "SlingshotAmmoBaseClass.generated.h"
+
+
+enum DamageType
+{
+	Physical,
+	Fire,
+	Water,
+	Poison,
+	Ice
+};
+
+UCLASS( ClassGroup=(Custom), meta=(BlueprintSpawnableComponent) )
+class STEVESLOTH_API USlingshotAmmoBaseClass : public UActorComponent
+{
+	GENERATED_BODY()
+
+public:	
+	// Sets default values for this component's properties
+	USlingshotAmmoBaseClass();
+	
+
+protected:
+	// Called when the game starts
+	virtual void BeginPlay() override;
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Name")
+	FString Name;
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Damage")
+	float Damage;
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Damage")
+	bool AreaOfEffect;
```

Comment:
Should also have an AOE damage amount
and the Radius of how far out the AOE goes from the center